### PR TITLE
Score redesign

### PR DESCRIPTION
**Pull request by SimplemathicDev**
Redesigned the score display, countdown and game over screens.
- Added text animations 🪄 to make it more engaging and intuitive
- Refactored some code in Main.cs
- Unified look (except the glow is missing from the buttons)

### CODE REVIEW

**Comment by elias-carlson** [comment]
holy shit this looks goooood. chef Levi been in the kitchennn👨‍🍳

**Comment by erikwessman** [comment]
giphy

LEVI RN FRRRRR

**Review by elias-carlson — APPROVED** [review]

File: weave/Scripts/Main.cs [hunk 1]
```diff
@@ -224,18 +248,27 @@ private void OnPlayerReachedGoal()
...
+    private void StartPreparationPhase()
+    {
         ClearLinesAndSegments();
         ClearAndSpawnGoals();
+        SetPlayerMovement(false);
+        _uiUpdateTimer.Timeout += UpdateCountdown;
+        _playerDelayTimer.Start();
+        _scoreDisplay.Enabled = false;
+
+        Task.Run(async delegate
```
Comment (theowiik) [inline weave/Scripts/Main.cs:265]: Could you do this logic with a timer? :)

**Comment by theowiik** [comment]
Looks amazing 🤤

File: weave/Scripts/Main.cs [hunk 2]
```diff
@@ -163,9 +181,15 @@ private void DetectPlayerOutOfBounds()
 
     private void GameOver()
     {
+        _gameIsRunning = false;
         _gameOverOverlay.Visible = true;
         _gameOverOverlay.FocusRetryButton();
-        ProcessMode = ProcessModeEnum.Disabled;
+        _musicPlayer.Stop();
```
Comment (theowiik) [inline weave/Scripts/Main.cs:187]: We will want to not stop the music player once you merge in my PR/main :)

File: weave/Scripts/Main.cs [hunk 3]
```diff
@@ -163,9 +181,15 @@ private void DetectPlayerOutOfBounds()
...
+        SetPlayerMovement(false);
+        SetPlayerTurning(false);
+        //ProcessMode = ProcessModeEnum.Disabled;
```
Comment (theowiik) [inline weave/Scripts/Main.cs:190]: Remove

File: weave/Scripts/ScoreDisplay.cs [hunk 4]
```diff
@@ -97,17 +102,29 @@ public void OnRoundComplete()
...
         }
 
-        _score += scoreIncrease * MathF.Pow(PlayerMultiplier, _playerCount - 1);
-
         _finishedRounds++;
         _timeSinceRoundStart = 0;
+
+        _animationPlayer.Play(name: "ScoreDisplayShine", customSpeed: 2.0f / WeaveConstants.CountdownLength);
+
+        Task.Run(async delegate
```
Comment (theowiik) [inline weave/Scripts/ScoreDisplay.cs:110]: Timer

File: weave/Scripts/Main.cs (hunk 1 above)
Comment (theowiik) [reply weave/Scripts/Main.cs:265]: image

**Review by theowiik — CHANGES_REQUESTED** [review]
🤤

File: weave/Scripts/Main.cs (hunk 2 above)
Comment (theowiik) [reply weave/Scripts/Main.cs]: resolved

File: weave/Scripts/Main.cs (hunk 3 above)
Comment (theowiik) [reply weave/Scripts/Main.cs]: resolved

File: weave/Scripts/ScoreDisplay.cs (hunk 4 above)
Comment (theowiik) [reply weave/Scripts/ScoreDisplay.cs]: resolved

**Pull request merged**